test/TestCases/Expressions/FastExp.cpp

```
@@ -0,0 +1,52 @@
+/*
+  ==============================================================================
+    DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+
+    Copyright 2023 by sonible GmbH.
+
+    This file is part of VCTR - Versatile Container Templates Reconceptualized.
+
+    VCTR is free software: you can redistribute it and/or modify
+    it under the terms of the GNU Lesser General Public License version 3
+    only, as published by the Free Software Foundation.
+
+    VCTR is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Lesser General Public License version 3 for more details.
+
+    You should have received a copy of the GNU Lesser General Public License
+    version 3 along with VCTR.  If not, see <https://www.gnu.org/licenses/>.
+  ==============================================================================
+*/
+
+#include <vctr_test_utils/vctr_test_common.h>
+
+template <vctr::is::realOrComplexFloatNumber T>
+T stdExp (T v) { return std::exp (v); }
+
+template <vctr::is::realOrComplexFloatNumber T>
```

oh so this works for complex numbers as well?

---

The SIMD implementations of FastExp limit it to floating types. However, when I change it to float here, the term `EqualsTransformedBy` fails to compile. I think since the simd implementation is for floating points only, so the complex test fall back to the sample wise calculation, which should work regardless.

---

Yeah that's to be expected, I just was surprised that the appxomiation calculation itself also works for complex numbers - which is actually probably not even that surprising thinking about it twice ;)

In that case I'd change my comment on constraining the actual expression to floating point and would now suggest to constrain it using is::realOrComplexFloatNumber

---

I think the approximation should even be reasonable in the complex plane around 0.